### PR TITLE
[ADF-4424] Pdf Viewer - password dialog does not close when pressing Escape

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

[ADF-4424](https://issues.alfresco.com/jira/browse/ADF-4424)

**What is the new behaviour?**

- closing the dialog without providing a password will return to the location from where the file was opened

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/viewer/components/pdfViewer.component.ts [hunk 1]
```diff
@@ -436,13 +438,14 @@ export class PdfViewerComponent implements OnChanges, OnDestroy {
...
                 data: { reason }
             })
             .afterClosed().subscribe((password) => {
-            if (password) {
-                callback(password);
-            }
+                if (password) {
+                    callback(password);
+                } else {
+                    this.location.back();
```
Comment (eromano) [inline lib/core/viewer/components/pdfViewer.component.ts]: @davidcanonieto has introduced a new service to go back in a better way can we use it https://github.com/Alfresco/alfresco-ng2-components/blob/development/lib/core/viewer/components/viewer.component.ts   ?  private previousRouteService: PreviousRouteService)

Comment (pionnegru) [reply lib/core/viewer/components/pdfViewer.component.ts]: doesn't seem to work.

Comment (eromano) [reply lib/core/viewer/components/pdfViewer.component.ts]: @davidcanonieto can you please check with @pionnegru if is possible to reuse your service or it needs some improvement?

File: lib/core/services/previous-route.service.ts [hunk 2]
```diff
@@ -39,4 +41,16 @@ export class PreviousRouteService {
...
+    public goBackToPreviousPage(): void {
+        if (this.previousUrl && this.previousUrl.includes('login') || window.history.length <= 2) {
+            this.goBackToHome();
+        } else {
+            this.location.back();
+        }
+    }
+
+    public goBackToHome(): void {
+        this.router.navigate([{outlets: {overlay: null, primary: ['home']}}]);
```
Comment (pionnegru) [inline lib/core/services/previous-route.service.ts]: what is 'home' from the perspective of another app?

Comment (pionnegru) [reply lib/core/services/previous-route.service.ts]: also 'login' route might not be also

Comment (DenysVuika) [reply lib/core/services/previous-route.service.ts]: @davidcanonieto yeah, for other apps outside demo shell there is no "home" route :)

Comment (pionnegru) [reply lib/core/services/previous-route.service.ts]: as suggested previously, navigation commands can give some details regarding location/overlay
```typescript
        const tree = this.router.parseUrl(this.router.url);
        const primaryGroup = tree.root.children[PRIMARY_OUTLET];
        const overlaySegmentGroup = tree.root.children['overlay'];

        console.log(overlaySegmentGroup.toString()); overlay -> /files/something/view
        console.log(primaryGroup.toString()); original location -> // files
```

Comment (eromano) [reply lib/core/services/previous-route.service.ts]: I agree need some love to make it generic can you please collaborate with David in order to improve it?

Comment (pionnegru) [reply lib/core/services/previous-route.service.ts]: I think the same problem can be found here 
https://github.com/Alfresco/alfresco-ng2-components/blob/development/lib/core/viewer/components/viewer.component.ts#L491

Comment (eromano) [reply lib/core/services/previous-route.service.ts]: yes is the same. for the login route check redirect we should use  :

```
 protected getLoginRoute(): string {
        return (
            this.appConfigService &&
            this.appConfigService.get<string>(
                AppConfigValues.LOGIN_ROUTE,
                'login'
            )
        );
    }
```

For home, we could define a similar strategy in the appconfig or we could just redirect the user to the '/'

Comment (DenysVuika) [reply lib/core/services/previous-route.service.ts]: Can we just pass the url as the route.data ? @pionnegru @eromano

Comment (pionnegru) [reply lib/core/services/previous-route.service.ts]: From my perspective, we are just trying to close the overlay and we can avoid checking url for login or do other logic if we are just navigate to primary outlet of the overlay.

Comment (DenysVuika) [reply lib/core/services/previous-route.service.ts]: I think this PR is turning into a mess. All we need is to close a password dialog upon pressing ESC.

Comment (eromano) [reply lib/core/services/previous-route.service.ts]: @DenysVuika why is a mess? I removed the routing from inside do you think is a bad idea?

Comment (DenysVuika) [reply lib/core/services/previous-route.service.ts]: @eromano no, that's a good idea :)

Comment (pionnegru) [reply lib/core/services/previous-route.service.ts]: with the latest commits the overlay is still there and you cannot navigate documentlist. are there any commits to this?

Comment (DenysVuika) [reply lib/core/services/previous-route.service.ts]: @pionnegru you might be looking at the outdated changes? I checked the code and there's no router there.

**Comment by pionnegru** [comment]
I test with lates commits and when the overlay viewer is closed in url you can still see /files(overlay:files/.../view) and you cannot interact with anything on the page

**Comment by eromano** [comment]
@pionnegru , Yep i see but should be it handle by the app, maybe listening on some close event?

**Comment by pionnegru** [comment]
Ok, I see now. We now only deal when its overlay/in document view but not as a secondary outlet

**Comment by DenysVuika** [comment]
What is the status of the PR?

**Comment by pionnegru** [comment]
the only tests failing are  process-services and process-services-cloud. do we expect this?

**Pull request merged**